pkg/server/router.go

```diff
@@ -85,6 +85,7 @@ func router(ctx context.Context, config *Config, cfg *cmds.Server) http.Handler
 	serverAuthed.Path(prefix + "/cert/cacerts").Handler(caCertReplaceHandler(serverConfig))
 	serverAuthed.Path("/db/info").Handler(nodeAuthed)
 	serverAuthed.Path(prefix + "/server-bootstrap").Handler(bootstrapHandler(serverConfig.Runtime))
+	serverAuthed.Path(prefix + "/token").Handler(tokenRequestHandler(ctx, serverConfig))
```

Comment:
Why do we need to expose this functionality over http? If I read the ADR correctly, the rotation can be triggered in two ways:
1 - k3s token rotate -t <OLD_TOKEN>
2 - k3s token rotate -t <OLD_TOKEN> --new-token <NEW_TOKEN>

In both cases, those commands are executed in the k3s server CLI

Reply:
The `k3s token rotate` command is just a wrapper around injecting http requests to the server. This allows the server to live operate on itself, instead of making the user shut down the cluster, run the command, then start the cluster back. You can run `k3s token rotate` and no workloads are affected. You can then restart the cluster with the new token later if you want to. This way is similar to how `k3s secrets-encrypt` operates.

Reply:
makes sense, thanks for the explanation!